File: src/main/java/de/schramm/royalbash/controller/service/core/effect/RemoveCreatureEffect.java
package de.schramm.royalbash.controller.service.core.effect;

import de.schramm.royalbash.controller.service.core.Context;
import de.schramm.royalbash.controller.service.core.Game;
import lombok.Builder;
import lombok.Value;
import lombok.val;

import java.util.Optional;

@Value
@Builder
public class RemoveCreatureEffect {

    public Game invoke(Context context) {

        val game = context.getGame();

        return context.getTargetCreature()
                .map(game::findCreature)
                .filter(Optional::isPresent)
                .map(Optional::get)
                .map(game::removeCreature)
                .orElse(game);
    }
}

File: src/main/java/de/schramm/royalbash/controller/service/core/effect/RemoveCreatureEffect.kt
package de.schramm.royalbash.controller.service.core.effect

import de.schramm.royalbash.controller.service.core.Context
import de.schramm.royalbash.controller.service.core.Game

class RemoveCreatureEffect {

    operator fun invoke(context: Context): Game {

        val game = context.game

        return context.getTargetCreature()
                .map { game.findCreature(it) }
                .filter { it.isPresent }
                .map { it.get() }
                .map { game.removeCreature(it) }
                .orElse(game)
    }
}
